### ISSUE TITLE
the old version cant be removed, update err!

### ISSUE THREAD

**Issue by countBunny** [intel/haxm#47]
![image](https://user-images.githubusercontent.com/17513986/39096265-2ec85454-4680-11e8-87ef-d3fd58751f60.png)
helper me, and this  bad version name is 6.0.3, if you can send me with a hax64.msi at that version, in china, it's hard to find a way to download this. hear for your email, thanks

**Comment by HaHoYou** [comment]
Here is it.  [haxm-windows_v6_0_3.zip](https://github.com/intel/haxm/files/1936760/haxm-windows_v6_0_3.zip)

**Comment by countBunny** [comment]
![image](https://user-images.githubusercontent.com/17513986/39135829-74b4b454-474c-11e8-9484-711041b4da3a.png)

the probless havent been finished. You should see clearly from the picture I provided, the windows installer need **haxm64.msi**, not the setup-package, this problem is *lead by System garbage cleaning* schedule, you put the origin file in the directory of AppDate's  Temp, it's easy to be deleted

**Comment by HaHoYou** [comment]
> Same problem for the 6.0.6 version

Have you tried new version in release page? Are you Windows OS? What is the log say?

**Comment by Fuzzyma** [comment]
I downloaded the newest version from the bwebsite which was 7.3. I am on win7. 
It tries to uninstall the old version but cannot find the installer (because its saved in the tempfolder - why???). So it asks for the path to the msi. But I couldnt trick it to use the 6.0.3 installer.

I saved the problem by deleting all registry keys with "Intel Hardware Acceleration Tool" which is the ugly way but it works

**Comment by HaHoYou** [comment]
I do not know the issue. But you could download latest, which is 7.3.2, and cover with latest, and uninstall the 7.3.2.

**Comment by Fuzzyma** [comment]
That does not work. I had 7.3.2.  It tries to uninstall the old version but cant. And then it stops.

The problem is the uninstall process!

**Comment by HaHoYou** [comment]
Ohh, I thought after your manual uninstalling, there should not be any problem. When met the error, any log? Is that error like "Failed to uninstall Intel HAXM! For details, please check the installation log xxx.log"?

**Comment by Fuzzyma** [comment]
I dont have any log. But LOTS OF people on the internet have exactly the same problem.

- The new installer tries to deinstall the old version
- the old installer cant be found (because its located in temp and that gets deleted by time)
- popup shows to locate the old installer -> thats why we need the old installers
- after cancelation the installation stops.
- new version NOT installed

I did not manually uninstall it. I deleted the registry entries. And thats a big hack!

**Comment by raphaelning** [comment]
@Fuzzyma Thanks for reporting this issue and sorry for the trouble. Before we root cause the missing MSI installer issue, the best we can do is upload to GitHub HAXM 6.0.6, and perhaps other old releases as well.

**Comment by Fuzzyma** [comment]
@raphaelning uploading the installers is a good first step. However, you would need to upload all versions because this problem is very old and it seems to happen with every version

**Comment by raphaelning** [comment]
Yes, we'll soon be adding a wiki page titled _HAXM 6.x Release Archive_, which complements our [Releases](https://github.com/intel/haxm/releases) page (7.0.0 and later, built from the open source codebase). Versions earlier than 6.0.0 should be extinct by now.

**Comment by Fuzzyma** [comment]
Sounds great!

**Comment by HaHoYou** [comment]
Here is https://github.com/intel/haxm/wiki/HAXM-6.x-Release-Archive .

**Comment by jrao1** [comment]
God, this bug is unbelievable, why does Intel design the installer to rely on temporary files in C:\Users\<Username>\AppData\Local\Temp? 

For the record, here's how I fixed it:
1. Try to uninstall it from Control Panel, the uninstall will fail but it will tell you where it expects the temporary files to be, record the directory, it's something like "C:\Users\<Username>\AppData\Local\Temp\Intel\HAXM\<Version>\<Old Date/Time>\"
2. Download the exact same version of HAXM, try to install it, it will fail too, but it should leave some temporary files under "C:\Users\<Username>\AppData\Local\Temp\Intel\HAXM\<Version>\<Today's Date/Time>"
3. Create the missing directory "C:\Users\<Username>\AppData\Local\Temp\Intel\HAXM\<Version>\<Old Date/Time>\", then copy all the files in "C:\Users\<Username>\AppData\Local\Temp\Intel\HAXM\<Version>\<Today's Date/Time>" to "C:\Users\<Username>\AppData\Local\Temp\Intel\HAXM\<Version>\<Old Date/Time>\"
4. Try to uninstall it from Control Panel again, it should work now.

**Comment by wcwang** [comment]
Thanks for your report. We have tried to reproduce your issue but cannot find it out. We will consider to redesign the uninstaller process to avoid such issue. Once the installer upgrades, we will announce it in our release notes. Thanks.
